src/Thruway/Role/Subscriber.php

```diff
@@ -128,6 +134,32 @@ protected function processSubscribed(ClientSession $session, SubscribedMessage $
             }
         }
     }
+    
+    /**
+     * Process unsubscribe error
+     * 
+     * @param \Thruway\AbstractSession $session
+     * @param \Thruway\Message\ErrorMessage $msg
+     */
+    protected function processUnsubscribeError(AbstractSession $session, ErrorMessage $msg)
+    {
+        foreach ($this->subscriptions as $key => $subscription) {
+            if ($subscription["unsubscribed_request_id"] === $msg->getErrorRequestId()) {
+                // reject the promise
+                $this->subscriptions[$key]['deferred']->reject($msg);
+                return;
+            }
+        }
+        
+        // Execution continues up here in case the original unsubscribe request has not been found
+        foreach ($this->unsubscriptionsPromises as $key => $unsubscriptionPromise) {
+            if ($unsubscriptionPromise["request_id"] === $msg->getErrorRequestId()) {
+                $unsubscriptionPromise["deferred"]->reject($msg);
+                unset($this->unsubscriptionPromises[$key]);
+                return;
+            }
+        }
+    }
```

Comment:
Does the spec say anything about what the client should do if there is an error unsubscribing?